app/models/category_list.rb

```diff
@@ -108,19 +108,28 @@ def find_categories
       category.notification_level = notification_levels[category.id] || default_notification_level
       category.permission = CategoryGroup.permission_types[:full] if allowed_topic_create.include?(category.id)
       category.has_children = category.subcategories.present?
+      category.subcategory_list = []
     end
 
     if @options[:parent_category_id].blank?
-      subcategories = {}
+      subcategory_ids = {}
+      subcategory_list = {}
       to_delete = Set.new
       @categories.each do |c|
         if c.parent_category_id.present?
-          subcategories[c.parent_category_id] ||= []
-          subcategories[c.parent_category_id] << c.id
+          subcategory_ids[c.parent_category_id] ||= []
+          subcategory_ids[c.parent_category_id] << c.id
+          subcategory_list[c.parent_category_id] ||= []
+          subcategory_list[c.parent_category_id] << c
```

Comment:
We should only run this if the option is present. Otherwise, we're adding more operations to the default action even if the option is not present and the array is not eventually used.